<WIP> Generate integrations

loki5265 opened the pull request:
* Add check dependencies

Fixes # .

Changes proposed in this pull request:
-
-

@anz-bank/sysl-developers

Commit:
<WIP> Generate integrations

* Add check dependencies
* Add find matching apps func
* Add find relevant apps func
* Add find integrations func
* Change dependency data structure
* Add stream lib to filter string slice
* Add entry function for ints
* Format ints file
* Add transformAttributes func to reduce code
* Fix comments

Commit:
<WIP> Generate integrations

* Add check dependencies
* Add find matching apps func
* Add find relevant apps func
* Add find integrations func
* Change dependency data structure
* Add stream lib to filter string slice
* Add entry function for ints
* Format ints file
* Add transformAttributes func to reduce code
* Fix comments

linwumeng commented on sysl2/sysl/ints.go:
```diff
@@ -0,0 +1,106 @@
+			calls := []*sysl.Call{}
+			makeCalls(endpoint.GetStmt(), calls)
+			for _, call := range calls {
+				targetName := formatAppName(call.GetTarget().GetPart())
```
3 level loop + large body is not good idea. Try to refactor to small pieces.

Commit:
Modify travis config and publish sysl2 image (#212)

* Modify repo name in travis configs
* Build image with sysl2 on travis
* Add sysl2 to /usr/bin/
* Add syslgen softlink to sysl2

Commit:
Bump version to 0.2.3

Commit:
Merge pull request #213 from anz-bank/release-v0.2.3

Bump version to 0.2.3

linwumeng commented on sysl2/sysl/ints.go:
```diff
@@ -0,0 +1,442 @@
+	return strings.Join(appNames, " :: ")
+}
+
+func validatePatterns(matched string, patterns []string) bool {
```
try implementing based on map. Prefer to hasPattern().

linwumeng commented on sysl2/sysl/ints.go:
```diff
@@ -0,0 +1,442 @@
+	return deps
+}
+
+func extractAppNames(dep *AppDependency) []string {
```
why not use dep as receiver?

linwumeng commented on sysl2/sysl/ints.go:
```diff
@@ -0,0 +1,442 @@
+	return apps
+}
+
+func extractEndpoints(dep *AppDependency) []string {
```
why not use dep as receiver?

linwumeng commented on sysl2/sysl/ints.go:
```diff
@@ -0,0 +1,442 @@
+	result := []string{}
+	m := arrayToMap(a)
+	for _, k := range b {
+		if _, ok := m[k]; ok {
```
prefer to has instead of ok

what if empty and nil cases?

linwumeng commented on sysl2/sysl/ints.go:
```diff
@@ -0,0 +1,442 @@
+	result := []string{}
+	m := arrayToMap(b)
+	for _, k := range a {
+		if _, ok := m[k]; !ok {
```
prefer to has instead of ok

what if empty and nil cases?

linwumeng commented on sysl2/sysl/ints.go:
```diff
@@ -0,0 +1,442 @@
+	return result
+}
+
+func isSub(child, parent []string) bool {
```
return len(substraction(child, parent)) == 0

what if empty and nil cases?

loki5265 replied:
add empty and nil cases handling

linwumeng commented on sysl2/sysl/ints.go:
```diff
@@ -0,0 +1,442 @@
+	return isSub
+}
+
+func contains(a string, arr []string) bool {
```
so why validatePatterns?

linwumeng commented on sysl2/sysl/ints.go:
```diff
@@ -0,0 +1,442 @@
+
+func findMatchingApps(module *sysl.Module, excludes []string, integrations []string, deps []*AppDependency) []string {
+	result := []string{}
+	appReStr := fmt.Sprintf(`^(?:%s)(?: *::|$)`, strings.Join(integrations, "|"))
```
extract to a function toPattern(comp []string) string.

linwumeng commented on sysl2/sysl/ints.go:
```diff
@@ -0,0 +1,442 @@
+	appReStr := fmt.Sprintf(`^(?:%s)(?: *::|$)`, strings.Join(integrations, "|"))
+	re := regexp.MustCompile(appReStr)
+	for _, apps := range deps {
+		appNames := extractAppNames(apps)
```
what if combine these two lines to one attaching to AppDependency. Then filter []AppDependency and map to []string.

linwumeng commented on sysl2/sysl/ints.go:
```diff
@@ -0,0 +1,442 @@
+	return result
+}
+
+func findApps(module *sysl.Module, excludes, matchingApps []string, deps []*AppDependency) []string {
```
refer to comments of previous function.

Commit:
Update gradle bintray plugin version

Gradle gives following error when publishing maven
'message:Unable to upload files: Maven group, artifact or version defined in the pom file do not match the file path'

The issue has been fixed from 1.8.1
https://github.com/bintray/gradle-bintray-plugin/issues/242#issuecomment-397826471

Commit:
Bump version to 0.2.4

Commit:
Inline docker script

sahejsingh committed:
Add NPM auth key (#217)

* Add NPM auth key
* USE npm publish token for deploy only

sahejsingh committed:
Fix ENEEDAUTH error: Add email to .npmrc (#218)

* Fix ENEEDAUTH error: Add email to .npmrc
* travis encrupt using -r anz-bank/sysl

Commit:
Bump version to 0.2.5

sahejsingh committed:
Esc unicode chars (#223)

* Add log level command line option
* 1) Allow \u unicode char in sysl strings, \n2) Allow reference in url params
* Add sourceContext to view
* Add test for log level

sahejsingh committed:
Release v0.2.6 (#224)

* Bump version to 0.2.6

* Remove github.com prefix from appveyor

sahejsingh committed:
Rest api param order (#225)

* Match param, query param order with legacy python

* Update package version to 1.0.4

sahejsingh committed:
Bump version to 0.2.7 (#226)

sahejsingh committed:
Add test to check gosysl windows binary (#227)

* Import paths need to have forward slash
* Add test to check gosysl windows binary
* Test gosysl binaries on travis

sahejsingh committed:
Release v0.2.8 (#228)

* Bump version to 0.2.8

* Update sysljs to 1.0.5

Commit:
<WIP> Generate integrations

* Add dependency set struct
* Add nil case handling for subtraction
* Rebase from the latest master branch

loki5265 committed:
Rollback urllib3 version (#230)

* Rollback version to 1.24.2

loki5265 committed:
Sysl generate sequence diagram (#209)

* Generate Sysl sequence diagrams (go)
